Assets/Scripts/PlayerController.cs

```diff
@@ -1,10 +1,85 @@
 using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.InputSystem;
 
 namespace SkyReach
 {
-    public class PlayerController : MonoBehaviour
+    [RequireComponent(typeof(Rigidbody2D))]
+    public class PlayerController : MonoBehaviour, Input.IPlayerActions
     {
+        private Rigidbody2D rigidBody;
+        private new Collider2D collider;
+
+        [SerializeField] private bool isSlippery = false;
+
+        private float raycastBuffer = 0.01f;
+        private Vector2 moveDirection;
+        [SerializeField] private float speed = 10.0f;
+        [SerializeField] private float jumpSpeed = 100000.0f;
+
+        private bool isGrounded = false;
+        private bool isJumping = false;
+
+        private Input input;
+
+
+        public void Awake()
+        {
+            rigidBody = GetComponent<Rigidbody2D>();
+            collider = GetComponent<Collider2D>();
+        }
+
+        public void OnEnable()
```

Comment:
What is the purpose of this Enable and Disable methods ?

Reply:
this is documented in https://github.com/pistolPeterson/Project-SkyReach/pull/7, it hooks to the events in the Input class